californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -792,10 +803,22 @@ private void sendNextBlock(final Exchange exchange, final Response response, fin
 			newSzx = status.getCurrentSzx();
 		}
 		int nextNum = status.getCurrentNum() + currentSize / newSize;
-		LOGGER.debug("sending next Block1 num={}", nextNum);
+		sendBlock(exchange, response, key, status, nextNum, newSzx);
+	}
+
+	private void SendBlockAgain(final Exchange exchange, final Response response, final KeyUri key,
+			final Block1BlockwiseStatus status) {
+		BlockOption block1 = response.getOptions().getBlock1();
+		int nextNum = status.getCurrentNum() * status.getCurrentSize() / block1.getSize();
+		sendBlock(exchange, response, key, status, nextNum, block1.getSzx());
```

Comment:
As already mentioned, I don't think, that changing the blocksize in the middle makes sense.
For me this is just a new transfer starting at 0.
(For block2 it is assumed, that the first too large block is processed, but the client want to indicate (late), that using a smaller blocksize is preferred. But that's different from 4.13).

Reply:
> (For block2 it is assumed, that the first too large block is processed, but the client want to indicate (late), that using a smaller blocksize is preferred. But that's different from 4.13).

I don't really get your point about block2, this also exist for block1 : 
```
CLIENT                                                     SERVER
     |                                                          |
     | CON [MID=1234], GET, /status                     ------> |
     |                                                          |
     | <------   ACK [MID=1234], 2.05 Content, 2:0/1/128        |
     |                                                          |
     | CON [MID=1235], GET, /status, 2:2/0/64           ------> |
     |                                                          |
     | <------   ACK [MID=1235], 2.05 Content, 2:2/1/64         |
     |                                                          |
     | CON [MID=1236], GET, /status, 2:3/0/64           ------> |
     |                                                          |
     | <------   ACK [MID=1236], 2.05 Content, 2:3/1/64         |
     |                                                          |
     | CON [MID=1237], GET, /status, 2:4/0/64           ------> |
     |                                                          |
     | <------   ACK [MID=1237], 2.05 Content, 2:4/1/64         |
     |                                                          |
     | CON [MID=1238], GET, /status, 2:5/0/64           ------> |
     |                                                          |
     | <------   ACK [MID=1238], 2.05 Content, 2:5/0/64         |

              Figure 4: Block-Wise GET with Late Negotiation
```

Reply:
Yes, the point is

> I'm currently not too deep in RFC 7959. As far as I remember:

Reply:
Just to mention: that changes the preferred blocksize at the begin.
That includes, that the server could process the large request, but prefers a smaller blocksize.
That uses 2.05 or 2.31, not 4.13.

Reply:
The figure 4 is for block2, or?

Reply:
oups sorry : bad copy/paste : 

```
  CLIENT                                                     SERVER
     |                                                          |
     | CON [MID=1234], PUT, /options, 1:0/1/128    ------>      |
     |                                                          |
     | <------   ACK [MID=1234], 2.31 Continue, 1:0/1/32        |
     |                                                          |
     | CON [MID=1235], PUT, /options, 1:4/1/32     ------>      |
     |                                                          |
     | <------   ACK [MID=1235], 2.31 Continue, 1:4/1/32        |
     |                                                          |
     | CON [MID=1236], PUT, /options, 1:5/1/32     ------>      |
     |                                                          |
     | <------   ACK [MID=1235], 2.31 Continue, 1:5/1/32        |
     |                                                          |
     | CON [MID=1237], PUT, /options, 1:6/0/32     ------>      |
     |                                                          |
     | <------   ACK [MID=1236], 2.04 Changed, 1:6/0/32         |

          Figure 9: Simple Atomic Block-Wise PUT with Negotiation
```